extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -562,67 +562,32 @@ public void accept(Throwable throwable) {
 
     private Uni<AuthorizationCodeTokens> refreshTokensUni(TenantConfigContext configContext, String refreshToken) {
 
-        return Uni.createFrom().emitter(new Consumer<UniEmitter<? super AuthorizationCodeTokens>>() {
-            @Override
-            public void accept(UniEmitter<? super AuthorizationCodeTokens> emitter) {
-                if (BlockingOperationControl.isBlockingAllowed()) {
-                    configContext.client.refreshToken(emitter, refreshToken);
-                } else {
-                    resolver.getBlockingExecutor().execute(new Runnable() {
-                        @Override
-                        public void run() {
-                            configContext.client.refreshToken(emitter, refreshToken);
-                        }
-                    });
-                }
-            }
-        });
+        if (BlockingOperationControl.isBlockingAllowed()) {
+            return configContext.provider.refreshTokens(refreshToken);
```

Comment:
[Optional]

you can write something like:

```
return configContext.provider.refreshTokens(refreshToken)
     .plug(u -> {
         if (! BlockingOperationControl.isBlockingAllowed()) {
             return u.runSubscriptionOn(resolver.getBlockingExecutor());
         } 
         return u;
     });
```

Reply:
@cescoffier Looks good to me - it feels more atomic this way, instead of having to duplicate the same call like `refreshTokens` twice, I'll give it a try

Reply:
This is done